app/autosynccontroller.cpp

```diff
@@ -8,33 +8,107 @@
  ***************************************************************************/
 
 #include "autosynccontroller.h"
+#include "coreutils.h"
 
-AutosyncController::AutosyncController( QObject *parent ) : QObject( parent )
-{
-  // raise NotImplemented :)
-}
+#include "qgsproject.h"
+#include "qgsvectorlayer.h"
 
-AutosyncController::~AutosyncController()
+AutosyncController::AutosyncController(
+  LocalProject *openedProject,
+  QgsProject *openedQgsProject,
+  QObject *parent
+)
+  : QObject( parent )
+  , mActiveProject( openedProject )
+  , mActiveQgsProject( openedQgsProject )
 {
-  // raise NotImplemented :)
+  if ( !mActiveQgsProject || !mActiveProject )
+  {
+    CoreUtils::log( QStringLiteral( "Autosync" ), QStringLiteral( "Received an invalid active project data" ) );
+    return;
+  }
+
+  // Register for data change of project's vector layers
+  for ( const QgsMapLayer *layer : mActiveQgsProject->mapLayers( true ) )
+  {
+    const QgsVectorLayer *vecLayer = qobject_cast<const QgsVectorLayer *>( layer );
+    if ( vecLayer )
+    {
+      QObject::connect( vecLayer, &QgsVectorLayer::afterCommitChanges, this, &AutosyncController::handleLocalChange );
+    }
+  }
+
+  if ( mActiveProject->localVersion < 0 )
```

Comment:
can we create function in ActiveProject "isMerginProject" that is called here? If you need to write comment in the code it is usually better to have function with the comment content...